File: sorting and searching/75. Sort Colors/sortColors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let zeros = 0, ones = 0, twos = 0;
    
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 0) zeros++;
        else if (nums[i] === 1) ones++;
        else if (nums[i] === 2) twos++;
    }

    for (let i = 0; i < zeros; i++) {
        nums[i] = 0;
    }
    for (let i = zeros; i < zeros + ones; i++) {
        nums[i] = 1;
    }
    for (let i = zeros + ones; i < nums.length; i++) {
        nums[i] = 2;
    }
};

File: sorting and searching/75. Sort Colors/sortColors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zeros = ones = twos = 0
        
        for i in range(len(nums)):
            if nums[i] == 0:
                zeros += 1
            if nums[i] == 1:
                ones += 1
            if nums[i] == 2:
                twos += 1
        
        nums[:zeros] = [0] * zeros
        nums[zeros:zeros + ones] = [1] * ones
        nums[zeros + ones:] = [2] * twos
        
        return nums